### ISSUE TITLE
Feature Request: Automatic embeds for x.com links

### ISSUE THREAD

**Issue by Jaccuse** [Automattic/jetpack#40180]
### Impacted plugin

Jetpack

### What

Automatically embed x.com links

### How

Currently, pasting an `x.com` link into a comment form gets you something like this:
![Image](https://github.com/user-attachments/assets/7f209502-8f49-4cf0-8146-c8b32b771790)

Some users are still using the Highlander comment editor (as opposed to Verbum) and would prefer `x.com` links to automatically embed, like this:
![Image](https://github.com/user-attachments/assets/5c7683e2-3574-45fb-9f8c-be4a59a5411f)

**Comment by jeherve** [comment]
@Automattic/vertex Do you think that's something we should implement in Verbum comments, or within the existing Twitter shortcode embed logic?
https://github.com/Automattic/jetpack/blob/3eeff5b7ea42234b79e617730dc07f075c256764/projects/plugins/jetpack/modules/shortcodes/tweet.php

For example, we handle Vimeo embeds in comments from the Vimeo shortcode:
https://github.com/Automattic/jetpack/blob/3eeff5b7ea42234b79e617730dc07f075c256764/projects/plugins/jetpack/modules/shortcodes/vimeo.php#L420

Related: #10682

**Comment by heavyweight** [comment]
Maybe a good start is to fix it with the embeds.
Posting this as comment: 
`https://twitter.com/goodside/status/1598129631609380864`
Will embed it for simple sites. I would expect the same to happen for x.com links.

If I'm not mistaken these don't work on atomic sites ATM. So the next pass would be to fix them there.

**Comment by jeherve** [comment]
> f I'm not mistaken these don't work on atomic sites ATM. So the next pass would be to fix them there.

That's correct. The embeds get turned into shortcodes that are displayed in comments via this mu-plugin:

fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sfubegpbqrf.cuc%3Se%3Q6412r3s9-og

We wouldn't be able to bring the entire mu-plugin to Jetpack just in one go, but maybe we should look at bringing the added functionality (embed to shortcode and hooking into `comment_text`) to the existing shortcode?

**Comment by jeherve** [comment]
> Maybe a good start is to fix it with the embeds.
> Posting this as comment:
> https://twitter.com/goodside/status/1598129631609380864
> Will embed it for simple sites. I would expect the same to happen for x.com links.

We would indeed have to update the embed to support the new domain. Although it is worth noting that the new domain isn't quite stable just yet. Related conversation:

https://core.trac.wordpress.org/ticket/59142

**Comment by supernovia** [comment]
I'm marking this as Triaged for now, but if it needs further PA involvement, send it back to "Needs Triage" and leave us a note!
